packages/framer-motion/src/animation/optimized-appear/start.ts

```diff
@@ -65,10 +68,26 @@ export function startOptimizedAppearAnimation(
          */
         window.MotionHandoffAnimation = handoffOptimizedAppearAnimation
 
-        window.MotionHasOptimisedTransformAnimation = (elementId?: string) => {
+        window.MotionHasOptimisedAnimation = (
+            elementId?: string,
+            valueName?: string
+        ) => {
             if (!elementId) return false
 
-            const animationId = appearStoreId(elementId, "transform")
+            /**
+             * Keep a map of elementIds that have started animating. We check
+             * via ID instead of Element because of hydration errors and
+             * pre-hydration checks. We also actively record IDs as they start
+             * animating rather than simply checking for data-appear-id as
+             * this attrbute might be present but not lead to an animation, for
+             * instance if the element's appear animation is on a different
+             * breakpoint.
+             */
+            if (!valueName) {
```

Comment:
In what cases do we not provide a value name for the optimized animation?

Reply:
In this instance it will check for *any* optimised animation - for instance when determining if we fire animations on layout effect or effect